### PR TITLE
Feature/seab 1401/collect emails

### PR DESCRIPTION
**Pull request by ByteMap**
Collect User Emails From GitHub

SEAB#1401

Implemented a getEmail() method that goes through every single email
from a GitHub user (regardless if they have private email turned on/off)
and returns their primary email

Created a integration test to verify that the fields of object Profile
is correct after updating the logged in user's metadata

Included a new fixture 'GitHubEmail.json' for Hoverfly

Implemented a new case for Hoverfly to check for the new endpoint
'/user/emails'

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 1]
```diff
@@ -895,7 +905,7 @@ public void syncUserMetadataFromGitHub(User user) {
             GHMyself myself = github.getMyself();
             User.Profile profile = new User.Profile();
             profile.name = myself.getName();
-            profile.email = myself.getEmail();
+            profile.email = getEmail(myself);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:908]: I wonder if to be safe, you shouldn't do `getEmail(myself) || myself.getEmail()`? Would that ever happen, that there's not a primary email, but the profile has an email?

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 2]
```diff
@@ -885,6 +886,15 @@ protected String getCommitID(String repositoryId, Version version) {
         return null;
     }
 
+    public String getEmail(GHMyself myself) throws IOException {
+        for (GHEmail email: myself.getEmails2()) {
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:890]: Does `getEmails2` always return a collection? Take a look at code to be sure. Concerned about possible NPE.

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java (hunk 1 above)
Comment (ByteMap) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:908]: Is it possible to have an email thats not primary? I thought GitHub automatically sets your only email as the primary email, or the first email you listed as the primary.

Comment (coverbeck) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:908]: I don't know, I'm just asking. :) Can you remove all your emails from your GitHub account, or does it require you to have at least one?

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java (hunk 2 above)
Comment (ByteMap) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:890]: From the documentation, getEmails2 always return a collection. The code throws a IOException, but I do not see a NPE being thrown in the code, or in the methods that it uses. This documentation says the method returns 'always non-null': https://github-api.kohsuke.org/apidocs/org/kohsuke/github/GHMyself.html#getEmails2()

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java (hunk 1 above)
Comment (ByteMap) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:908]: GitHub requires me to have at least one email. I am unable to remove all of my emails.

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 3]
```diff
@@ -885,6 +886,15 @@ protected String getCommitID(String repositoryId, Version version) {
         return null;
     }
 
+    public String getEmail(GHMyself myself) throws IOException {
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:889]: Is this public for a reason?

Comment (ByteMap) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:889]: No reason at all, nothing outside of the class is using the getEmail() method. Is it best to change it to private?

Comment (ByteMap) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java]: Changed it to private.

Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java]: Yeah, just makes things easier to reason about. 
e.g. when refactoring, I know I can change this method without breaking anything outside of this class, the IDE can tell me when the method is un-used, etc.

**Pull request merged**